deps/rabbitmq_peer_discovery_aws/src/rabbit_peer_discovery_aws.erl

```diff
@@ -347,23 +351,66 @@ get_hostname_by_tags(Tags) ->
             Names
     end.
 
--spec get_hostname_path() -> path().
-get_hostname_path() ->
-    UsePrivateIP = get_config_key(aws_use_private_ip, ?CONFIG_MODULE:config_map(?BACKEND_CONFIG_KEY)),
-    HostnamePath = get_config_key(aws_hostname_path, ?CONFIG_MODULE:config_map(?BACKEND_CONFIG_KEY)),
-    FinalPath = case HostnamePath of
-        ["privateDnsName"] when UsePrivateIP -> ["privateIpAddress"];
-        P -> P
+
+
+-spec get_hostname_paths() -> [path()].
+get_hostname_paths() ->
+    M = ?CONFIG_MODULE:config_map(?BACKEND_CONFIG_KEY),
+    UsePrivateIP = get_config_key(aws_use_private_ip, M),
+    
```

Comment:
There is extraneous whitespace here - in the diff, I see four space characters here.